datamap/mv/core/src/main/scala/org/apache/carbondata/mv/rewrite/DefaultMatchMaker.scala

```diff
@@ -127,8 +126,27 @@ object SelectSelectNoChildDelta extends DefaultMatchPattern with PredicateHelper
     }
   }
 
-  def apply(
-      subsumer: ModularPlan,
+  private def isLeftJoinView(subsumer: ModularPlan): Boolean = {
+    if (subsumer.isInstanceOf[modular.Select]) {
+      val sel = subsumer.asInstanceOf[modular.Select]
```

Comment:
better to use match clause to simply the logic in this function

Reply:
ok